checkST2110.js

```diff
@@ -325,20 +403,69 @@ const test_20_71_1 = (sdp, params) => {
   let streams = sdp.split(/[\r\n]m=/);
   let errors = [];
   if (params.videoOnly) {
-    for ( let s = 1 ; s < streams.length ; s++ ) {
+    for (let s = 1; s < streams.length; s++) {
       if (!streams[s].startsWith('video')) {
         errors.push(new Error(`Stream ${s}: Media type is not 'video' and video only files are in test, as per SMPTE 2110-20 Section 7.1.`));
       }
     }
+  } else {
+    console.log("Test Skipped: Test ST 2110-20 Section 7.1 Test 1 - Use --videoOnly to check for all video.");
   }
+
+  if (params.videoOnly && params.verbose && errors.length == 0)
+    console.log("Test Passed: Test ST 2110-20 Section 7.1 Test 1 - All streams are video");
+
   return errors;
 };
 
+// Test ST 2110-10 Section 6,2 Test 1 - For all video streams, check video params
+const test_10_62_1 = (sdp, params) => {
+  let errors = [];
+  let lines = splitLines(sdp);
+  videoPresent = false;
+
+  for (let x = 0; x < lines.length; x++) {
+    if (!lines[x].startsWith('m=video')) {
+      continue;
+    }
+    videoPresent = true;
+    let videoMatch = lines[x].match(videoPattern);
+    if (!videoMatch) {
+      errors.push(new Error(`Line ${x + 1}: Found a media description for video with a pattern that is not acceptable.`));
+      continue;
+    }
+    // Check port number - SMPTE 2110-10 Section 6.2 says shall be UDP, so assume 0-65535
+    let port = +videoMatch[1];
+    if (isNaN(port) || port < 0 || port > 65535) {
+      errors.push(new Error(`Line ${x + 1}: RTP video stream description with invalid port '${port}', with reference to ST 2110-10 Section 6.2 'shall use UDP'.`));
+    }
+    // Check RTP type - SMPTE 2110-10 Section 6.2 says shall be RTP, no allowance for SRTP
+    if (videoMatch[3] === 'RTP/SAVP') {
+      errors.push(new Error(`Line ${x + 1}: SRTP protocol is not allowed by SMPTE ST 2110-10 Section 6.2.`));
+    }
+    // Check dynamic range - assume 2110-20 is always dynamic
+    let payloadType = +videoMatch[4];
+    if (isNaN(payloadType) || payloadType < 96 || payloadType > 127) {
+      errors.push(new Error(`Line ${x + 1}: Dynamic payload type expected for SMPTE 2110-defined video.`));
+    }
+  }
+
+  if (videoPresent == false)
+    errors.push(new Error(`m=video expected for SMPTE 2110-defined video.`));
```

Comment:
If this is a test case for ST 2110-10, why should it only be applied for video SDP files? ST 2110-30, etc., also refer to ST 2110-10.

Reply:
Ah, I guess my point here was the duplication with `test_30_62_2`...

Reply:
Also, I now notice that `allSections` adds all the ST 2110-30 tests, via `section_30_62`, for media type `raw`. Was that your intent? They don't cause any failures for a valid ST 2110-20 SDP file I think, but in `--verbose` mode we see that they've been run.

Similarly, I'm not sure if it's a problem that `allSections` doesn't distinguish between "Audio or other", as again this path includes `section_30_62` as well as the ST 2110-10 basics...

Reply:
Looking at this I realized that there was some refactoring needed to clean this up and use media type and encoding name from the spec consistently.  I should have been using media type as either video or audio and encoding name as raw or jxsv then setting up test set accordingly. I made those changes and committed. Should be clearer now and less duplication and easier to extend for h.264/h.265 and other new codecs.